atanh(tanh(x)) is not simplified

Test case
```
x = symbols('x', real=True)
print( atanh(tanh(x)).simplify() )
```
evaluates to `atanh(tanh(x))`

Hello,

I have run the full test suite on my local machine and I'm seeing a number of failures. I believe these are related to my local environment and not to my changes, as the specific bug fix I implemented appears to be working correctly.

Could you please advise on how to resolve these test failures?

Here is the log:   =============================================== short test summary info ================================================
FAILED sympy/codegen/tests/test_algorithms.py::test_newtons_method_function__rtol_cse_nan - assert 2.86126401550282484098850784656751e-15 < 2.64523030128337532347187054305861e-15
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_multiple_paths_from_non_root[paths0-expected_paths0] - AssertionError: assert 'doc/src/modules/utilities' in 'C:\\Users\\chris\\OneDrive\\Desktop\\sympy\\doc\\src\\modules...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths0] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths1] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths2] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths3] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths4] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths0] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths1] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths2] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths3] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_multiple_keywords - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_keyword_match_in_multiple_files - AssertionError: assert 'sympy/integrals/tests/test_heurisch.py::test_heurisch_symbolic_coeffs_1130' in 'C:\\Users\\c...       
FAILED sympy/utilities/tests/test_mathml.py::test_xxe -   File "<string>", line 3
==== 14 failed, 13128 passed, 387 skipped, 203 deselected, 333 xfailed, 2 xpassed, 7 warnings in 1568.57s (0:26:08) ====

Hello,

I have run the tests using `pytest sympy` as you suggested. The same test failures occurred.

Here is the log:   
 =============================================== short test summary info ================================================
FAILED sympy/codegen/tests/test_algorithms.py::test_newtons_method_function__rtol_cse_nan - assert 2.86126401550282484098850784656751e-15 < 2.64523030128337532347187054305861e-15
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_multiple_paths_from_non_root[paths0-expected_paths0] - AssertionError: assert 'doc/src/modules/utilities' in 'C:\\Users\\chris\\OneDrive\\Desktop\\sympy\\doc\\src\\modules...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths0] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths1] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths2] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths3] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_string_as_keyword[paths4] - AssertionError: assert 'sympy/physics/mechanics/tests/test_kane3.py::test_bicycle' in 'C:\\Users\\chris\\OneDrive\\D...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths0] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths1] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths2] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_integer_as_keyword[paths3] - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_multiple_keywords - AssertionError: assert 'sympy/core/tests/test_sympify.py::test_issue_3538' in 'C:\\Users\\chris\\OneDrive\\Desktop\\...
FAILED sympy/testing/tests/test_runtests_pytest.py::TestUpdateArgsWithPaths::test_keyword_match_in_multiple_files - AssertionError: assert 'sympy/integrals/tests/test_heurisch.py::test_heurisch_symbolic_coeffs_1130' in 'C:\\Users\\c...       
FAILED sympy/utilities/tests/test_mathml.py::test_xxe -   File "<string>", line 3
==== 14 failed, 13128 passed, 387 skipped, 203 deselected, 333 xfailed, 2 xpassed, 8 warnings in 1426.60s (0:23:46) ====

--- a/.mailmap
+++ b/.mailmap
@@ -1794,4 +1794,4 @@ zzj <29055749+zjzh@users.noreply.github.com>
 Łukasz Pankowski <lukpank@o2.pl>
 彭于斌 <1931127624@qq.com>
 袁野 (Yuan Ye) <yuanyelele@tutanota.com>
-cp50 <chrispaul68002@gmail.com>
+Chris Paul <chrispaul68002@gmail.com> cp50 <chrispaul68002@gmail.com>

--- a/.mailmap
+++ b/.mailmap
@@ -1794,4 +1794,4 @@ zzj <29055749+zjzh@users.noreply.github.com>
 Łukasz Pankowski <lukpank@o2.pl>
 彭于斌 <1931127624@qq.com>
 袁野 (Yuan Ye) <yuanyelele@tutanota.com>
-Chris Paul <chrispaul68002@gmail.com> cp50 <chrispaul68002@gmail.com>
+cp50 <chrispaul68002@gmail.com>

--- a/.mailmap
+++ b/.mailmap
@@ -1701,6 +1701,7 @@ carstimon <carstimon@gmail.com>
 ck Lux <lux.r.ck@gmail.com>
 cocolato <haiizhu@outlook.com> Hai Zhu <35182391+cocolato@users.noreply.github.com>
 codecruisader <nnisarg55@gmail.com>
+cp50 <chrispaul68002@gmail.com>
 cym1 <16437732+cym1@users.noreply.github.com>
 damianos <damianos@semmle.com>
 dandiez <47832466+dandiez@users.noreply.github.com>

--- a/.mailmap
+++ b/.mailmap
@@ -1794,3 +1794,4 @@ zzj <29055749+zjzh@users.noreply.github.com>
 Łukasz Pankowski <lukpank@o2.pl>
 彭于斌 <1931127624@qq.com>
 袁野 (Yuan Ye) <yuanyelele@tutanota.com>
+Chris Paul <chrispaul68002@gmail.com> cp50 <chrispaul68002@gmail.com>

--- a/.mailmap
+++ b/.mailmap
@@ -470,6 +470,7 @@ Chiu-Hsiang Hsu <wdv4758h@gmail.com>
 Chris Cameron <chrisjamescameron1@gmail.com> ccam80 <63381855+ccam80@users.noreply.github.com>
 Chris Conley <chrisconley15@gmail.com>
 Chris Kerr <chris.kerr@mykolab.ch>
+Chris Paul <chrispaul68002@gmail.com> cp50 <chrispaul68002@gmail.com>
 Chris Smith <smichr@gmail.com>
 Chris Smith <smichr@gmail.com> <Donna Smith@.(none)>
 Chris Swierczewski <cswiercz@gmail.com>
@@ -1794,4 +1795,3 @@ zzj <29055749+zjzh@users.noreply.github.com>
 Łukasz Pankowski <lukpank@o2.pl>
 彭于斌 <1931127624@qq.com>
 袁野 (Yuan Ye) <yuanyelele@tutanota.com>
-Chris Paul <chrispaul68002@gmail.com> cp50 <chrispaul68002@gmail.com>

--- a/AUTHORS
+++ b/AUTHORS
@@ -1377,3 +1377,4 @@ Mathis Cros <mathis.cros@telecom-paris.fr>
 Arnav Mummineni <45217840+RCoder01@users.noreply.github.com>
 Thangaraju Sibiraj <85477603+t-sibiraj@users.noreply.github.com>
 KJaybhaye <krushnajaybhaye01@gmail.com>
+cp50 <chrispaul68002@gmail.com>

--- a/sympy/functions/elementary/hyperbolic.py
+++ b/sympy/functions/elementary/hyperbolic.py
@@ -1652,19 +1652,20 @@ class atanh(InverseHyperbolicFunction):
         if arg.is_zero:
             return S.Zero
 
-        if isinstance(arg, tanh) and arg.args[0].is_number:
+        if isinstance(arg, tanh):
             z = arg.args[0]
             if z.is_real:
                 return z
-            r, i = match_real_imag(z)
-            if r is not None and i is not None:
-                f = floor(2*i/pi)
-                even = f.is_even
-                m = z - I*f*pi/2
-                if even is True:
-                    return m
-                elif even is False:
-                    return m - I*pi/2
+            if z.is_number:
+                r, i = match_real_imag(z)
+                if r is not None and i is not None:
+                    f = floor(2*i/pi)
+                    even = f.is_even
+                    m = z - I*f*pi/2
+                    if even is True:
+                        return m
+                    elif even is False:
+                        return m - I*pi/2
 
     @staticmethod
     @cacheit

--- a/sympy/functions/elementary/tests/test_hyperbolic.py
+++ b/sympy/functions/elementary/tests/test_hyperbolic.py
@@ -1106,6 +1106,7 @@ def test_atanh():
     assert atanh(tanh(-3 + 7*I)) == -3 - 2*I*pi + 7*I
     assert atanh(tanh(9 - I*2/3)) == 9 - I*2/3
     assert atanh(tanh(-32 - 123*I)) == -32 - 123*I + 39*I*pi
+    assert atanh(tanh(x, evaluate=False)) == x
 
 
 def test_atanh_rewrite():

--- a/sympy/functions/elementary/tests/test_hyperbolic.py
+++ b/sympy/functions/elementary/tests/test_hyperbolic.py
@@ -1106,7 +1106,8 @@ def test_atanh():
     assert atanh(tanh(-3 + 7*I)) == -3 - 2*I*pi + 7*I
     assert atanh(tanh(9 - I*2/3)) == 9 - I*2/3
     assert atanh(tanh(-32 - 123*I)) == -32 - 123*I + 39*I*pi
-    assert atanh(tanh(x, evaluate=False)) == x
+    x = Symbol('x', real=True)
+    assert atanh(tanh(x)) == x
 
 
 def test_atanh_rewrite():

